core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -242,6 +255,7 @@ public static TableMetadata read(FileIO io, InputFile file) {
     }
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
I remember this also shows up when adding primary key #2010 , I think we should probably find some way to refactor the code a bit.

Reply:
Yeah I imagine there would be some conflict when either of the PRs got merged. I think the main problem with refactoring this is that a lot of the things it parsed should be parsed together/validate each other; e.g. parse of schema array has to come together with current schema id since both will either present at the same time, or derived from a different field, and same goes for sort order/partition spec. Since Java only allows returning a single object per method, we either have to refactor the code a bit and ensure the logic flow doesn't change and is readable, or let the new methods to return things like `tuple<Integer, List<Schema>>` that I'm not sure which way is cleaner.

Reply:
I think we could have a method that parses schemas and a separate one that returns the current schema ID. That said, I'm fine with suppressing the complexity warning for now.